sklearn/metrics/tests/test_common.py

```diff
@@ -415,17 +418,17 @@ def test_symmetry():
                     msg="%s seems to be symmetric" % name)
 
 
-@ignore_warnings
-def test_sample_order_invariance():
+@pytest.mark.parametrize(
+        'name',
+        set(ALL_METRICS) - METRIC_UNDEFINED_BINARY_MULTICLASS)
+def test_sample_order_invariance(name):
     random_state = check_random_state(0)
     y_true = random_state.randint(0, 2, size=(20, ))
     y_pred = random_state.randint(0, 2, size=(20, ))
     y_true_shuffle, y_pred_shuffle = shuffle(y_true, y_pred, random_state=0)
 
-    for name, metric in ALL_METRICS.items():
-        if name in METRIC_UNDEFINED_BINARY_MULTICLASS:
-            continue
-
+    with ignore_warnings():
```

Comment:
Just curious, did you find a problem with the interaction of ignore_warnings and parametrize? I bumped into some quirks as I indicated in https://github.com/scikit-learn/scikit-learn/pull/11151#issuecomment-395712587.

Reply:
Yes, I think same things as what you mentioned in your comment, if I recall correctly,
 * `ignore_warning` (as decorator) + parametrize failed on Python 2
 * `ignire_warning` (as decorator or not) + pytest.mark.filterwarnings didn't seem to work (but that might be fixed in #11151)